lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2746,15 +2746,25 @@ namespace {
         return nullptr;
       }
 
+      auto dc = decl->getDeclContext();
       // Bail if this is `std::chrono::tzdb`. This type causes issues in copy
       // constructor instantiation.
       // FIXME: https://github.com/apple/swift/issues/73037
-      if (decl->getDeclContext()->isNamespace() &&
-          decl->getDeclContext()->getParent()->isStdNamespace() &&
+      if (dc->isNamespace() && dc->getParent()->isStdNamespace() &&
           decl->getIdentifier() &&
           (decl->getName() == "tzdb" || decl->getName() == "time_zone_link"))
         return nullptr;
 
+      // Import the C++ swift::String as the swift String type.
+      if (dc->isNamespace() &&
+          cast<clang::NamespaceDecl>(dc)->getName() == "swift" &&
+          decl->getIdentifier() && decl->getName() == "String") {
+        auto swiftDecl = Impl.SwiftContext.getStringDecl();
+        Impl.SwiftContext.registerExportedClangDecl(swiftDecl,
+                                                    decl->getCanonicalDecl());
```

Comment:
Is this state change on import going to be susceptible to ordering issues? For example, could we end up querying the ClangTypeConverter for Swift's `String` without having seen the C++ `swift::String`? It might be more robust if `ClangTypeConverter::convert` were able to call into the Clang importer to ask "Do you have a mapping for this type?".

Reply:
I think this should not happen now. We populate this map when we import the declarations and query it during lowering. Currently, `ClangTypeConverter::convert` should handle missing mappings gracefully. But I can move the code around if you think it would be more straightforward that way.